### PR TITLE
[18] Write unit test for TakeSurveyModel, GetSurveyDetailUseCase, SubmitSurveyUseCase

### PR DESCRIPTION
**Pull request by manh-t**
https://github.com/manh-t/flutter-survey/issues/18

## What happened 👀

Write unit test for TakeSurveyModel, GetSurveyDetailUseCase, SubmitSurveyUseCase

## Insight 📝

I forgot to push the PR 😓 

## Proof Of Work 📹

CI passed

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

**Comment by sleepylee** [comment]
please rebase 🙋

**Comment by manh-t** [comment]
@sleepylee all rebased 🙇

File: test/pages/takesurvey/take_survey_model_test.dart [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+import 'package:flutter_survey/pages/takesurvey/take_survey_state.dart';
+import 'package:flutter_survey/usecase/base/base_use_case.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../mock/mock_data.mocks.dart';
+import '../../utils/survey_util.dart';
+import '../../utils/test_util.dart';
+
+class TakeSurveyModelListener extends Mock {
```
Comment (sleepylee) [inline test/pages/takesurvey/take_survey_model_test.dart:17]: I'd prefer creating these mock implementations somewhere else, a new file

**Review by sleepylee — APPROVED** [review]
just a minor comment; the PR is kinda big and it packs multiple things to review. Please separate them next time for each class...

**Comment by manh-t** [comment]
@sleepylee sorry, I'll notice that 🙏 Updated 🚀

**Pull request merged**